core/src/main/java/org/infinispan/container/DefaultDataContainer.java

```diff
@@ -553,8 +553,9 @@ public void executeTask(final KeyFilter<? super K> filter, final BiConsumer<? su
       if (action == null)
          throw new IllegalArgumentException("No action specified");
 
+      long now = timeService.wallClockTime();
       entries.forEach((K key, InternalCacheEntry<K, V> value) -> {
-         if (filter.accept(key)) {
+         if (filter.accept(key) && !value.isExpired(now)) {
```

Comment:
Good catch! I would say lets create another PR for the `DefaultDataContainer` as I see it is also broken for compute method as well. We should make sure to check all the other methods.